alchemy/src/scope.ts

```diff
@@ -537,6 +550,49 @@ export class Scope {
     return promise;
   }
 
+  /**
+   * Install global process exit handlers to ensure cleanup
+   */
+  private installExitHandlers() {
+    if (this.parent) return;
+    onExit((code) => {
+      if (this.cleanups.length === 0) return;
+      this.logger.log(kleur.gray("Exiting..."));
+      Promise.allSettled(this.cleanups.map((cleanup) => cleanup())).then(() => {
+        console.log("Exiting with code:", code); // this never gets logged, thanks Miniflare
+        process.exit(code);
```

Comment:
why do we need to call this?

Reply:
If you return `true` (as we do on line 565), that tells the `onExit` function that we'll call process.exit, so don't exit for us. This is important because otherwise, it'll exit before those cleanup promises can be awaited (`onExit` doesn't let you return a promise).

The debug log I added there isn't 100% true - it doesn't get logged if Miniflare is there intercepting the exit signal (which it shouldn't, I can make an issue for that tomorrow), but if we don't have Miniflare messing up our process (e.g. if your app is just a Vite website or something), then we do want to handle the exit ourselves.